solutions/usaco-839.mdx

```diff
@@ -0,0 +1,57 @@
+---
+id: usaco-839
+source: USACO Gold 2018 Open
+title: Talent Show
+author: Arpan Banerjee
+---
+
+[Official Editorial](http://www.usaco.org/current/data/sol_talent_gold_open18.html)
+
+<Warning title="Common Misconception">
+Instead of using binary search as the official solution does, a knapsack can be done. An array of talents can be kept, where $\texttt{talent}[i]$ is the minimum weight required to create a talent sum of $i$. To make sure each cow is used at most once and the order of the cows doesn't matter, the indices of $\texttt{talent}$ are iterated though from greatest to least, and the $\texttt{cows}$ are iterated though in the outer loop; the implementation should make this clear. This maximizes the ratio $\frac{t}{w}$ as desired. Then, the maximum ratio of $\frac{t_i}{w_i}$ can be easily found by going through $\texttt{talent}$ and ignoring any $w_i$ that is less than $W$. 
+
+My $\mathcal{O}(\text{Total talent } \cdot n)$ code is below:
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include<bits/stdc++.h>
+#define w first
+#define t second
+using namespace std;
+
+const int inf=2e9;
+const int total_talent=250000;
+int n,w; double ans=0;
+
+signed main(){
+       ifstream cin("talent.in"); ofstream cout("talent.out");
+       cin >> n >> w;
```

Comment:
the spacing is inconsistent; either add spaces around all operators or don't put any spaces (most don't have spaces but this line and line 39 are different)